ludwig/data/preprocessing.py

```diff
@@ -1383,7 +1454,8 @@ def preprocess_for_training(
         skip_save_processed_input=False,
         preprocessing_params=default_preprocessing_parameters,
         backend=LOCAL_BACKEND,
-        random_seed=default_random_seed
+        random_seed=default_random_seed,
+        callbacks=[]
```

Comment:
In general, we should avoid using `[]` as a default argument in Python, as it becomes a mutable constant, which can have unexpected side effects when you call the function multiple times. Better alternative is to use `None`.